### PR TITLE
Release 3.8.1 CHANGELOG and INSTALL Updates

### PR DESCRIPTION
**Pull request by kmuriki**
Release 3.8.1 CHANGELOG and INSTALL file updates.

Also merges everything from the master branch into release-3.8.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
-  - _Migration guidance_
-  - _New features / functionalities_
+## v3.8.1 - [2021-08-12]
 
+### Changed defauls / behaviours
 
-_The old changelog can be found in the `release-2.6` branch_
+  - `build-singularity` example pulls the base image from Docker/RockyLinux 
+    instead of Docker/CentOS
+  - Set LABELs properly in the container when pulling from Docker/OCI source.
```
Comment (DrDaveD) [inline CHANGELOG.md]: I believe this change is the one that was undone, not in this release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
-  - _Deprecated / removed commands_
-  - _Changed defaults / behaviors_
-  - _Migration guidance_
-  - _New features / functionalities_
+## v3.8.1 - [2021-08-12]
 
+### Changed defauls / behaviours
 
-_The old changelog can be found in the `release-2.6` branch_
+  - `build-singularity` example pulls the base image from Docker/RockyLinux
```
Comment (DrDaveD) [inline CHANGELOG.md]: This is too trivial, I would leave it out.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+## v3.8.1 - [2021-08-12]
 
+### Changed defauls / behaviours
 
-_The old changelog can be found in the `release-2.6` branch_
+  - `build-singularity` example pulls the base image from Docker/RockyLinux 
+    instead of Docker/CentOS
+  - Set LABELs properly in the container when pulling from Docker/OCI source.
+    Use --force in a build to override a label that already exists in source.
+  - Partially replace the github.com/satori/go.uuid module.
```
Comment (DrDaveD) [inline CHANGELOG.md]: This also doesn't affect anything in the behavior so it isn't needed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
+# Singularity Changelog
 
-  - _Renamed commands_
-  - _Deprecated / removed commands_
-  - _Changed defaults / behaviors_
-  - _Migration guidance_
-  - _New features / functionalities_
+## v3.8.1 - [2021-08-12]
 
+### Changed defauls / behaviours
```
Comment (DrDaveD) [inline CHANGELOG.md]: This is misspelled but the whole section is getting deleted anyway.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,22 +1,32 @@
...
+  - `build-singularity` example pulls the base image from Docker/RockyLinux 
+    instead of Docker/CentOS
+  - Set LABELs properly in the container when pulling from Docker/OCI source.
+    Use --force in a build to override a label that already exists in source.
+  - Partially replace the github.com/satori/go.uuid module.
 
+### Bug Fixes
 
-# v3.8.0 - [2021-06-15]
+  - Allow escaped `\$` in a SINGULARITYENV_ var to set a literal `$` in
```
Comment (DrDaveD) [inline CHANGELOG.md:7]: Also please add a note for changed functionality in #6008.  It's in some ways similar to this first note so I would put it first or second.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

File: internal/pkg/runtime/engine/singularity/process_linux.go [hunk 6]
```diff
@@ -631,18 +632,22 @@ func injectEnvHandler(senv map[string]string) interpreter.OpenHandler {
 			`
 			b.WriteString(fmt.Sprintf(defaultPathSnippet, env.DefaultPath))
 
+			// https://github.com/sylabs/singularity/issues/43
```
Comment (ikaneshiro) [inline internal/pkg/runtime/engine/singularity/process_linux.go]: Do we want this reference in the codebase?

File: e2e/env/regressions.go [hunk 7]
```diff
@@ -99,3 +99,26 @@ func (c ctx) issue5057(t *testing.T) {
 		e2e.ExpectExit(0),
 	)
 }
+
+// https://github.com/sylabs/singularity/issues/43
```
Comment (ikaneshiro) [inline e2e/env/regressions.go]: And this

**Review by ikaneshiro — COMMENTED** [review]
Apologies, I missed some initially on the first pass.

**Review by ikaneshiro — APPROVED** [review]

File: internal/pkg/runtime/engine/singularity/process_linux.go (hunk 6 above)
Comment (DrDaveD) [reply internal/pkg/runtime/engine/singularity/process_linux.go]: I don't think it hurts, it's a valid URL that explains the issue

**Review by DrDaveD — APPROVED** [review]

**Pull request merged**